### PR TITLE
Various bug fixes related to resuming workflows

### PR DESCRIPTION
**Pull request by scottfrazer**

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 1]
```diff
@@ -455,8 +456,8 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
 
     jesJobSetup match {
       case Failure(ex) =>
-        log.error(s"Failed to create a JES run", ex)
-        FailedExecutionHandle(ex)
+        log.warn(s"Failed to create a JES run", ex)
+        throw ex
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: could perhaps use a comment similar to that above

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala (hunk 1 above)
Comment (scottfrazer) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: will do

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 2]
```diff
@@ -243,6 +243,7 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
 
     backendCall.instantiateCommand match {
       case Success(command) => runWithJes(backendCall, command, jesInputs, jesOutputs, runIdForResumption)
+      case Failure(ex: SocketTimeoutException) => throw ex // probably a GCS transient issue, throwing will cause it to be retried
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Hmm, something seems odd about throwing an Exception from inside a Future {...} block but if this is the idiom then no problem.

Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: So, a Future.failed means "retry" but a FailedExecutionHandle means "don't retry"? Could we have a RetryMeExecutionHandle or is that too awkward?

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 3]
```diff
@@ -455,8 +456,8 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
 
     jesJobSetup match {
       case Failure(ex) =>
-        log.error(s"Failed to create a JES run", ex)
-        FailedExecutionHandle(ex)
+        log.warn(s"Failed to create a JES run", ex)
+        throw ex  // Probably a transient issue, throwing retries it
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Same comment as above re throwing exception as an indicator for "retry me please" feeling strange.

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala [hunk 4]
```diff
@@ -654,7 +654,7 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
     // These executions have no corresponding recorded operation ID and are therefore not resumable.
     def collectNonResumableDatabaseKeys(executionsAndJobs: Seq[(Execution, JesJob)]): Seq[ExecutionDatabaseKey] = {
       executionsAndJobs collect {
-        case (execution, job) if execution.status.toExecutionStatus == ExecutionStatus.Running && job.jesJobId.isEmpty =>
+        case (execution, job) if execution.status.toExecutionStatus == ExecutionStatus.Running && job.jesId.isEmpty =>
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Ah, yes, I can see why this was causing problems :)

Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Too late now maybe but these names are far too similar for comfort.

File: src/main/scala/cromwell/logging/TerminalLayout.scala [hunk 5]
```diff
@@ -20,8 +20,13 @@ class TerminalLayout extends LayoutBase[ILoggingEvent] {
...
     val timestamp = new SimpleDateFormat("YYYY-MM-dd HH:mm:ss,SS").format(new Date())
 
+    val ansiColor = Option(System.getProperty("RAINBOW_UUID")) match {
+      case Some(_) => "UUID\\((.*?)\\)".r.findFirstMatchIn(event.getFormattedMessage).map(s => (Math.abs(17 * s.group(1).map(_.toInt).product) % 209) + 22).getOrElse(2)
+      case None => 2
+    }
+
     val highlightedMessage = event.getFormattedMessage
-      .replaceAll("UUID\\((.*?)\\)", TerminalUtil.highlight(2, "$1"))
+      .replaceAll("UUID\\((.*?)\\)", TerminalUtil.highlight(ansiColor, "$1"))
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/logging/TerminalLayout.scala]: A wild terminal layout change has appeared!

Is this a duplicate of Scott's branch or genuinely meant to be here?

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala (hunk 2 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Yeah I agree this should be documented more.  The `FailedExecutionHandle` means the fate of the execution is definite and it is failed.  The failed `Future` is interpreted by the caller of this method as a condition which merits a retry.

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Probably more of question for @mcovarr, though I too find the different between throw and message passing a bit confusing but perhaps there's a good reason that I'm not remembering

File: src/main/scala/cromwell/logging/TerminalLayout.scala (hunk 5 above)
Comment (cjllanwarne) [inline src/main/scala/cromwell/logging/TerminalLayout.scala]: Oops! I thought this was one of Miguel's PRs for some reason. Ignore this comment!

Comment (scottfrazer) [inline src/main/scala/cromwell/logging/TerminalLayout.scala]: This is my branch :)

Yeah, this is a purely opt-in thing.  You won't notice any difference at all unless you want to see it.

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala (hunk 4 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: I certainly wouldn't object to renaming this here and in the database, something like jesRunId maybe?

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: I like jesRunId

Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Yup.

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Alright changed the column to JES_RUN_ID and jesRunId in scala land

File: src/main/migrations/changesets/rename_jes_id.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet author="sfrazer" id="rename-jes-id-to-jes-run-id">
+        <renameColumn columnDataType="INT"
```
Comment (cjllanwarne) [inline src/main/migrations/changesets/rename_jes_id.xml]: This was a VARCHAR(100) before!

Comment (scottfrazer) [inline src/main/migrations/changesets/rename_jes_id.xml]: oh boo... I was too hasty

Comment (cjllanwarne) [inline src/main/migrations/changesets/rename_jes_id.xml]: More evidence these names were too similar before!

**Comment by cjllanwarne** [comment]
Once the JES_ID/JES_RUN_ID type is sorted, :+1:

File: src/main/migrations/changesets/rename_jes_id.xml (hunk 6 above)
Comment (scottfrazer) [inline src/main/migrations/changesets/rename_jes_id.xml]: :)  I'm just testing out the DB changes now then I'll merge

**Pull request merged**